src/somesy/core/models.py

```diff
@@ -350,6 +358,31 @@ def full_name(self) -> str:
 
         return " ".join(names) if names else ""
 
+    def to_name_email_string(self) -> str:
+        """Convert project metadata person object to poetry string for person format `full name <x@y.z>`."""
+        return f"{self.full_name} <{self.email}>"
+
+    @classmethod
+    def from_name_email_string(cls, person: str) -> Person:
+        """Return a `Person` based on an name/e-mail string like `full name <x@y.z>`.
+
+        If the name is `A B C`, then `A B` will be the given names and `C` will be the family name.
+        """
+        m = re.match(r"\s*([^<]+)<([^>]+)>", person)
+        names, mail = (
+            list(map(lambda s: s.strip(), m.group(1).split())),
+            m.group(2).strip(),
+        )
+        # NOTE: for our purposes, does not matter what are given or family names,
+        # we only compare on full_name anyway.
+        return Person(
+            **{
+                "given-names": " ".join(names[:-1]),
+                "family-names": names[-1],
+                "email": mail,
+            }
+        )
+
```

Comment:
I refactored these out into the core model, because they are now used by both poetry and the julia thing and I guess its common enough that it might come up more times and we should not duplicate it.